bin/varnishncsa/varnishncsa.c

```diff
@@ -163,8 +163,42 @@ static struct ctx {
 	int			recv_compl;
 } CTX;
 
+enum format_policy {
+        FMTPOL_FIRST = 0,
+        FMTPOL_REQ,
+        FMTPOL_RESP,
+};
+
 static void parse_format(const char *format);
 
+static unsigned
+process_format_policy(enum format_policy fp, unsigned first)
```

Comment:
I requested that we pass the fragment instead of a boolean in https://github.com/varnishcache/varnish-cache/pull/4213#discussion_r1824213963 so the function signature should look like this:

```c
static unsigned
process_format_policy(enum format_policy fp, const struct fragment *frag);
```

And since we operate on a fragment, we can make it fit better with the rest of the code:

```c
static unsigned
frag_needed(const struct fragment *frag, enum format_policy fp);
```

This way we can completely centralize the policy, and call sites are easier to read too.